src/cli.zig

```diff
@@ -511,7 +511,13 @@ pub const Arguments = struct {
         opts.env_files = args.options("--env-file");
         opts.extension_order = args.options("--extension-order");
 
-        ctx.passthrough = args.remaining();
+        {
+            var list = std.ArrayList(string).init(allocator);
```

Comment:
There's a very common performance mistake in this code.

1. Creating an array of known length
2. Appending to it item-by-item

This means it will resize potentially multiple times.
and it will append to it only one at a time, which is much slower than a single memcpy.

Instead, we can do

```suggestion
            const positional = args.positionals();
            const remaining = args.remaining();
            var list = std.ArrayList(string).initCapacity(allocator, positional.len + remaining.len);
            list.appendSliceAssumeCapacity(positional);
            list.appendSliceAssumeCapacity(remaining);
            ctx.passthrough = list.items;
```